pytradfri/command.py

```diff
@@ -1,13 +1,14 @@
 """Command implementation."""
 from __future__ import annotations
 
+from collections.abc import Callable
 from copy import deepcopy
-from typing import Any, Callable, Optional
+from typing import Any, Generic, TypeVar
 
-TypeProcessResultCb = Optional[Callable[[Any], Any]]
+T = TypeVar("T")
 
 
-class Command:
+class Command(Generic[T]):
```

Comment:
I have seen `T = TypeVar("T")` but never used it. Just did some reading of the concept and am I correct in understanding it as this? By using it, we can say that a method returns a `Command` of type `Group`. That automatically types the `process_result` as a callable taking argument `Group `and that the result is also of type `Group`?

Reply:
It sort of helps us do dynamic typing?

Reply:
`process_result` will return an instance of type T, yes.

It has two benefits as I see it.

1. It'll be easier to see what a Command class will return after the api has executed it. (this PR) 
2. We can use the TypeVar T of the command in the api and type the return value of the request method so that the request method will show the correct return value for any typed command we pass. (the follow up PR)

Reply:
I don't follow you completely but look forward to seeing the next PR to see what you mean.